### PR TITLE
chore(rest): added validation to OAuth token URL

### PR DESCRIPTION
**Pull request by igpetrov**
chore(rest): added validation to OAuth token URL

### CODE REVIEW

File: connectors/http/graphql/element-templates/graphql-outbound-connector.json [hunk 1]
```diff
@@ -193,11 +193,15 @@
...
     "id" : "authentication.oauthTokenEndpoint",
-    "label" : "Oauth token endpoint",
+    "label" : "OAuth 2.0 token endpoint",
     "description" : "The OAuth token endpoint",
     "optional" : false,
     "constraints" : {
-      "notEmpty" : true
+      "notEmpty" : true,
+      "pattern" : {
+        "value" : "^(=|http://|https://|secrets|\\{\\{).*$",
```
Comment (johnBgood) [inline connectors/http/graphql/element-templates/graphql-outbound-connector.json:202]: I could use `secretsssss` for instance, or `http://///dldl`, not sure we want a full validation though so feel free to ignore my comment.

File: connectors/http/graphql/element-templates/graphql-outbound-connector.json [hunk 2]
```diff
@@ -193,11 +193,15 @@
...
+        "message" : "Must be a http(s) URL"
```
Comment (johnBgood) [inline connectors/http/graphql/element-templates/graphql-outbound-connector.json:203]: could be http URL, secret, expression etc

**Review by johnBgood — APPROVED** [review]
Minor comments only, great that you refactored the code 👍

File: connectors/http/graphql/element-templates/graphql-outbound-connector.json (hunk 1 above)
Comment (igpetrov) [reply connectors/http/graphql/element-templates/graphql-outbound-connector.json:202]: I guess, we're fine with that. We discussed the same issue with generic URL validation but decided not to go over-strict

Comment (igpetrov) [reply connectors/http/graphql/element-templates/graphql-outbound-connector.json:202]: I will create a ticket to revisit the URL pattern

Comment (igpetrov) [reply connectors/http/graphql/element-templates/graphql-outbound-connector.json:202]: There it is https://github.com/camunda/connectors/issues/2236

**Pull request merged**